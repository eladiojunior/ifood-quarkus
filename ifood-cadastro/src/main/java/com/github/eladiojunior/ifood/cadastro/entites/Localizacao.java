package com.github.eladiojunior.ifood.cadastro.entites;

import io.quarkus.hibernate.orm.panache.PanacheEntityBase;

import javax.persistence.*;

@Entity
@Table(name = "TB_LCLZO")
public class Localizacao extends PanacheEntityBase {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ID_LCLZO")
    public Long id;

    @Column(name = "CD_LTTDE")
    public Double latitude;

    @Column(name = "ID_LNGTE")
    public Double longitude;

}
