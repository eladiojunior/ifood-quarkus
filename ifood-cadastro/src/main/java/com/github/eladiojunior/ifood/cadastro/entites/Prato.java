package com.github.eladiojunior.ifood.cadastro.entites;

import io.quarkus.hibernate.orm.panache.PanacheEntityBase;

import javax.persistence.*;
import java.math.BigDecimal;

@Entity
@Table(name = "TB_PRATO")
public class Prato extends PanacheEntityBase {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ID_PRATO")
    public Long id;

    @Column(name = "NM_PRATO")
    public String nome;

    @Column(name = "DS_PRATO")
    public String descricao;

    @Column(name = "VL_PRATO")
    public BigDecimal preco;

    @ManyToOne
    public Restaurante restaurante;

}
