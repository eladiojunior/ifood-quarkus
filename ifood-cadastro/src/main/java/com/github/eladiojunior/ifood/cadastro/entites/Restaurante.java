package com.github.eladiojunior.ifood.cadastro.entites;

import io.quarkus.hibernate.orm.panache.PanacheEntityBase;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import javax.json.bind.annotation.JsonbDateFormat;
import javax.persistence.*;
import java.util.Date;

@Entity
@Table(name = "TB_RSTRE")
public class Restaurante extends PanacheEntityBase {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ID_RSTRE")
    public Long id;

    @Column(name = "NM_PRPRO_RSTRE")
    public String proprietario;

    @Column(name = "NM_RSTRE")
    public String nome;

    @Column(name = "NR_CNPJ_RSTRE")
    public String cnpj;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "ID_LCLZO" )
    public Localizacao localizacao;

    @Column(name = "DH_RGSTO_RSTRE")
    @CreationTimestamp
    @JsonbDateFormat(value = "yyyy-MM-dd")
    public Date dataHoraRegistro;

    @Column(name = "DH_ATLZO_RSTRE")
    @UpdateTimestamp
    @JsonbDateFormat(value = "yyyy-MM-dd")
    public Date dataHoraAtualizacao;

}
