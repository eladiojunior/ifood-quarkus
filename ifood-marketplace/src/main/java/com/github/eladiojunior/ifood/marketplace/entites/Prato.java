package com.github.eladiojunior.ifood.marketplace.entites;
import java.math.BigDecimal;

public class Prato {
    private Long id;
    private String nome;
    private String descricao;
    private BigDecimal preco;
    private Restaurante restaurante;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getDescricao() {
        return descricao;
    }

    public void setDescricao(String descricao) {
        this.descricao = descricao;
    }

    public BigDecimal getPreco() {
        return preco;
    }

    public void setPreco(BigDecimal preco) {
        this.preco = preco;
    }

    public Restaurante getRestaurante() {
        return restaurante;
    }

    public void setRestaurante(Restaurante restaurante) {
        this.restaurante = restaurante;
    }
}
