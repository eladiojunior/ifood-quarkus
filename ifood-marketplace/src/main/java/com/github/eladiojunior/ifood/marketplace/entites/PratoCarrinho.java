package com.github.eladiojunior.ifood.marketplace.entites;

public class PratoCarrinho {
    private String cliente;
    private Long idPrato;

    public String getCliente() {
        return cliente;
    }

    public void setCliente(String cliente) {
        this.cliente = cliente;
    }

    public Long getIdPrato() {
        return idPrato;
    }

    public void setIdPrato(Long idPrato) {
        this.idPrato = idPrato;
    }
}
