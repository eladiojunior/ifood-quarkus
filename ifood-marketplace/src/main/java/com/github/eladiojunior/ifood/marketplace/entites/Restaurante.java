package com.github.eladiojunior.ifood.marketplace.entites;

public class Restaurante {
    private Long id;
    private String nome;
    private Localizacao localizacao;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public Localizacao getLocalizacao() {
        return localizacao;
    }
    public void setLocalizacao(Localizacao localizacao) {
        this.localizacao = localizacao;
    }

    @Override
    public String toString() {
        return "Restaurante{" +
                "id=" + id +
                ", nome='" + nome + '\'' +
                ", localizacao=" + localizacao +
                '}';
    }
}
