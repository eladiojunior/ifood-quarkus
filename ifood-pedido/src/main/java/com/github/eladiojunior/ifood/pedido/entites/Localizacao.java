package com.github.eladiojunior.ifood.pedido.entites;

public class Localizacao {
    private Double latitude;
    private Double longitude;

    public Double getLatitude() {
        return latitude;
    }

    public void setLatitude(Double latitude) {
        this.latitude = latitude;
    }

    public Double getLongitude() {
        return longitude;
    }

    public void setLongitude(Double longitude) {
        this.longitude = longitude;
    }
}
