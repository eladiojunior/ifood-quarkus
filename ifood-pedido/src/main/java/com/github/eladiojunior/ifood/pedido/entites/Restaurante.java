package com.github.eladiojunior.ifood.pedido.entites;

public class Restaurante {
    private String nome;
    private Localizacao localizacao;

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public Localizacao getLocalizacao() {
        return localizacao;
    }

    public void setLocalizacao(Localizacao localizacao) {
        this.localizacao = localizacao;
    }
}
