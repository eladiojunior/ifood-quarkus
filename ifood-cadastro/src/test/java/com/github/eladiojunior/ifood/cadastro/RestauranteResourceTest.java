package com.github.eladiojunior.ifood.cadastro;

import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;
import com.github.eladiojunior.ifood.cadastro.entites.Restaurante;
import com.github.eladiojunior.ifood.cadastro.util.TokenUtils;
import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import io.restassured.RestAssured;
import io.restassured.http.ContentType;
import io.restassured.http.Header;
import io.restassured.specification.RequestSpecification;
import org.approvaltests.Approvals;
import org.junit.Assert;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import javax.ws.rs.core.Response;

@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
@QuarkusTestResource(CadastroTestLifecycleManager.class)
public class RestauranteResourceTest {

    private String token;

    @BeforeEach
    public void gerarToken() {
        try {
            token = TokenUtils.generateTokenString("/JWTProprietarioClaims.json", null);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    @Test
    @DataSet("restaurantes-cenario-01.yml")
    public void testListarRestaurantes() {
        String resultado = given()
          .when().get("/restaurantes")
          .then()
             .statusCode(200)
             .extract().asString();
        Approvals.verifyJson(resultado);
    }

    /**
     * Especificar o tipo do dados (JSON) utilizado nas requisições de teste.
     * @return
     */
    private RequestSpecification given() {
        return RestAssured.given().contentType(ContentType.JSON).header(new Header("Authorization", "Bearer " + token));
    }

    @Test
    @DataSet("restaurantes-cenario-01.yml")
    public void testAtualizarRestaurantes() {
        Restaurante dto = new Restaurante();
        dto.nome = "Alteração no nome do restaurante";
        Long idParamRestaurante = 1L;

        given()
                .with().pathParam("id", idParamRestaurante)
                .body(dto)
                .when().put("/restaurantes/{id}")
                .then()
                .statusCode(Response.Status.NO_CONTENT.getStatusCode())
                .extract().asString();
        Restaurante findById = Restaurante.findById(idParamRestaurante);
        Assert.assertEquals(dto.nome, findById.nome);
    }

}