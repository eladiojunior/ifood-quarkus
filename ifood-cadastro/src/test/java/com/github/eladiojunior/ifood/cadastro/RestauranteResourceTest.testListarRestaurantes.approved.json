[
  {
    "id": 1,
    "nome": "Nome do restaurante 1",
    "proprietario": "Nome do proprietario 1"
  },
  {
    "id": 2,
    "nome": "Nome do restaurante 2",
    "proprietario": "Nome do proprietario 2"
  }
]